#! /bin/sh -e
## octave-2.1.53-demos.dpatch by Rafael Laboissiere <rafael@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Use warn_fortran_indexing instead of do_fortran_indexing

[ -f debian/patches/00patch-opts ] && . debian/patches/00patch-opts
patch_opts="${patch_opts:--f --no-backup-if-mismatch}"

if [ $# -ne 1 ]; then
  echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
  exit 1
fi
case "$1" in
  -patch)
    patch $patch_opts -p1 < $0;;
  -unpatch)
    patch $patch_opts -p1 -R < $0;;
  *)
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1;;
esac

exit 0
--- plplot-orig/bindings/octave/PLplot/support/__pl_matstr.m
+++ plplot/bindings/octave/PLplot/support/__pl_matstr.m
@@ -34,10 +34,10 @@
 	out(nn,:) = "";
       endif
     endif
-    old_dofi = do_fortran_indexing;
-    do_fortran_indexing = 1;
+    old_dofi = warn_fortran_indexing;
+    warn_fortran_indexing = 0;
     out(toascii (out) == 0) = " ";
-    do_fortran_indexing = old_dofi;
+    warn_fortran_indexing = old_dofi;
   else
     help __pl_matstr
   endif
